### PR TITLE
Done with EKSU, fixed all Logo URLS

### PR DESCRIPTION
**Pull request by stephengade**
This commit attends to the following issues:
- #23 
- #22 

I also completed EKSU Faculties and departments too

### CODE REVIEW

File: pages/api/v1/universities/index.ts [hunk 1]
```diff
@@ -3,36 +3,80 @@ import fs from "fs";
...
-            return JSON.parse(fileContents);
-          });
-
-        res.status(200).json(universities);
-      } catch (error) {
-        res.status(500).json({
-          message: "Error retrieving university data",
-          error: error,
+        const universities = loadUniversities();
+        const { type, city, state, acronym, page = "1", limit = "20" } = query;
```
Comment (kaf-lamed-beyt) [inline pages/api/v1/universities/index.ts]: great work @stephengade! 

let's filter by university fullname too

File: pages/api/v1/universities/index.ts [hunk 2]
```diff
@@ -3,36 +3,80 @@ import fs from "fs";
 import path from "path";
 import { University } from "@/types/university.type";
 
+let cachedUniversities: University[] | null = null;
+
+function loadUniversities(): University[] {
```
Comment (kaf-lamed-beyt) [inline pages/api/v1/universities/index.ts:8]: walk me through why you have this @stephengade.

**Review by kaf-lamed-beyt — CHANGES_REQUESTED** [review]
Beautiful work you've done so far @stephengade! Thank you so much!

I left a question and requested one change. Kindly take a look.

File: pages/api/v1/universities/index.ts (hunk 2 above)
Comment (stephengade) [reply pages/api/v1/universities/index.ts:8]: `let cachedUniversities: University[] | null = null` 

This is just to store universities' data once it's loaded - `null`, the initial stage when the data hasn't been loaded

`loadUniversities`  function that will load the universities' data. before it loads the data, it should first check if the data has already been cached in cachedUniversities

If the data is already cached, the function can return it directly without needing to read it all over again

PS: Better caching mechanism can be used

Comment (kaf-lamed-beyt) [reply pages/api/v1/universities/index.ts:8]: we'll figure out a better caching approach as we progress. This should work fine for now. 

Thanks again!

**Review by kaf-lamed-beyt — APPROVED** [review]

**Pull request merged**